src/runtime/builtin_modules/builtins.cpp

```diff
@@ -2381,6 +2386,7 @@ void setupBuiltins() {
         = BoxedClass::create(type_cls, object_cls, 0, 0, sizeof(Box), false, "ellipsis", false, NULL, NULL, false);
     ellipsis_cls->giveAttr("__repr__",
                            new BoxedFunction(BoxedCode::create((void*)ellipsisRepr, STR, 1, "ellipsis.__repr__")));
+    ellipsis_cls->tp_repr = ellipsis_repr;
```

Comment:
The proper fix here is to call `ellipsis_cls->freeze()` (and get rid of `ellipsis_repr`), which will automatically fill in tp_repr for us.